core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -417,16 +418,19 @@ private ClusterCacheStatus initCacheStatusIfAbsent(String cacheName) {
       return cacheStatusMap.computeIfAbsent(cacheName, (name) -> {
          // We assume that any cache with partition handling configured is already defined on all the nodes
          // (including the coordinator) before it starts on any node.
-         Configuration cacheConfiguration = cacheManager.getCacheConfiguration(cacheName);
          AvailabilityStrategy availabilityStrategy;
-         if (cacheConfiguration != null && cacheConfiguration.clustering().partitionHandling().enabled()) {
+         Configuration cacheConfiguration = cacheManager.getCacheConfiguration(cacheName);
+         PartitionHandling  partitionHandling = cacheConfiguration != null ? cacheConfiguration.clustering().partitionHandling().whenSplit() : null;
+         boolean resolveConflictsOnMerge = partitionHandling == PartitionHandling.ALLOW_READ_WRITES && cacheConfiguration.clustering().partitionHandling().mergePolicy() != null;
```

Comment:
You can likely get conflicts even without allowing writes: primary owner applies the write but backup (in the other partition) does not get them. The application will get an exception, but it does not have an easy way to schedule the resolution when it will be possible.